processing/src/main/java/io/druid/query/Query.java

```diff
@@ -65,6 +66,8 @@
 
   boolean hasFilters();
 
+  DimFilter getFilter();
```

Comment:
suggestion :  change return type to Optional<DimFilter> since not all queries have filters.